stylix/mk-target.nix

```diff
@@ -0,0 +1,77 @@
+{
+  # name used for creation of options
+  name,
+
+  # name used for description of options
+  humanName,
+
+  # condition for automatically enabling this target
+  autoEnable ? true,
+
+  # options to be added under `options.stylix.targets.${name}`
+  extraOptions ? { },
+
+  # list of functions with args representing Stylix options; function results
+  # only merged if all relevant Stylix options are not null
+  configElements ? [ ],
+
+  # function with args representing Stylix options; function result always
+  # merged
+  generalConfig ? null,
+}:
+# Note: we only return a plain module so that calling code can use module args, while using this as their module body
+let
+  configModule =
+    { config, lib, ... }:
+    let
+      cfg = config.stylix.targets.${name};
+      functionArgNames =
+        fn:
+        lib.pipe fn [
+          lib.functionArgs
+          builtins.attrNames
+        ];
+      getStylixAttrs =
+        fn:
+        lib.genAttrs (functionArgNames fn) (
+          arg:
+          if arg == "cfg" then
+            cfg
+          else if arg == "colors" then
+            config.lib.stylix.${arg}
+          else
+            config.stylix.${arg}
+              or (throw "stylix: Stylix arg was requested by mk-target but does not exist: ${arg}")
+        );
+      mkConfig = fn: fn (getStylixAttrs fn);
+      mkConditionalConfig =
+        fn:
+        let
+          attrs = functionArgNames fn;
+        in
+        lib.mkIf (builtins.all (attr: attr != null) attrs) (fn (getStylixAttrs fn));
```

Comment:
I don't understand in what scenario one of the attrNames could be `null`.

These names come from `functionArgNames`, which returns the `attrNames` list of the function args set.

`builtins.attrNames` cannot return a null list item, elements in the list will all be strings.

Some comments and/or more descriptive names may help understand the intent, but I suspect something is broken here.

Reply:
Perhaps the intent was to check the actual option values, rather than the option names?